rustworkx-core/src/dag_algo.rs

```diff
@@ -313,6 +316,115 @@ where
     Ok(Some((path, path_weight)))
 }
 
+/// Return a list of graph layers
+///
+/// A layer is a subgraph whose nodes are disjoint, i.e.,
+/// a layer has depth 1. The layers are constructed using a greedy algorithm.
+///
+/// Arguments:
+///
+/// * `graph` - The graph to get the layers from
+/// * `first_layer` - A list of node ids for the first layer. This
+///     will be the first layer in the output
+///
+/// ```
+/// use rustworkx_core::petgraph::prelude::*;
+/// use rustworkx_core::dag_algo::layers;
+/// use rustworkx_core::dictmap::*;
+///
+/// let edge_list = vec![
+///  (0, 1),
+///  (1, 2),
+///  (2, 3),
+///  (3, 4),
+/// ];
+///
+/// let graph = DiGraph::<u32, u32>::from_edges(&edge_list);
+/// let layers: Vec<Vec<usize>> = layers(&graph, vec![0,1]).unwrap();
+/// let expected_layers = vec![vec![0,1], vec![1,2], vec![2,3], vec![3,4], vec![4]];
+/// assert_eq!(layers, expected_layers)
+/// ```
+pub fn layers<G>(graph: G, first_layer: Vec<usize>) -> Result<Vec<Vec<usize>>, LayersError>
```

Comment:
I was comparing your PR with #1186 and noticed that even though we both return node ids, in this case you return node ids as a `Vec<Vec<usize>>` instead of a `Vec<Vec<G::NodeId>>`. Was there any particular reason for this choice? I based mine on the `longest_path` function, which also returned a `Vec<Vec<G::NodeId>>`.

Reply:
In this case, I don't think there is a particular reason, might just be a case of me following the previous code a little bit too tightly.

From what I can see it's just so we can send a regular `Vec` with `usize` values, since it might be a bit easier for the user to deal with, but then again, it might work the same way with `G::NodeId`. I will get back to you on this.